src/text-editor.coffee

```diff
@@ -892,7 +892,7 @@ class TextEditor extends Model
   # Determine whether the user should be prompted to save before closing
   # this editor.
   shouldPromptToSave: ({windowCloseRequested, projectHasPaths}={}) ->
-    if windowCloseRequested and projectHasPaths
+    if windowCloseRequested and projectHasPaths and atom.stateStore.isConnected
```

Comment:
We've been trying to avoid `atom` global references in `TextEditor` to facilitate extraction when the time comes. Could you inject the state store as a dependency or find some way to avoid this global reference? I'm not sure where we reference it downstream from this but if that's also a global reference, then maybe ignore my advice.